targeteditor/src/main/scala/explore/targeteditor/TargetQueries.scala

```diff
@@ -145,4 +144,35 @@ object TargetQueries {
       )(value)
   }
 
+  /**
+   * Updates all the fields of sideral tracking
+   */
+  def updateSiderealTracking(
+    t: SiderealTracking
+  ): IndexedStateT[Eval, EditSiderealInput, EditSiderealInput, Unit] = {
```

Comment:
Why not `State[EditSiderealInput, Unit]`?

Reply:
Right, I should change that

Reply:
A downside of this method is that we change every field. This is fine for the response of a simbad query.
But if we edit a single field, should we only set that field on the save?

Reply:
If a single field is edited, it'd be nice to only set that field because then the undoer will weave that change with other possible remote changes. In other words: it will keep remote changes to other fields that can happen between the set and the undo. Otherwise, the undo will override those remote changes and take the whole target to the state it was before the initial local editing.

Reply:
That's a good point, but I'm inclined to think it doesn't make too much of a difference in practice.  It's more a convenience for people using the API than anything that the non-updated fields can be empty.

Reply:
> I'm inclined to think it doesn't make too much of a difference in practice

I was referring there to the backend and not taking into account the impact on undo that @rpiaggio mentions.  I see what you're saying but now I'm trying to decide if, from the user perspective, it is actually good that the entire target changes as a unit.  At least you would get a consistent collection of values?

Reply:
I think it is in principle ok to send everything if only one field change, though I have to think about undo.
I was a bit more worried about the extra traffic and db activity on updating fields that don't change.

Reply:
I think we should send everything after a catalog query. After all, values could have changed remotely on the server since we last read them, and in this case we do want to override everything with the values from the catalog.